components/event-bus/cmd/mesh-namespaces-migration/serviceinstance.go

```diff
@@ -43,14 +55,38 @@ func (m *serviceInstanceManager) populateServiceInstances(namespaces []string) e
 		if err != nil {
 			return errors.Wrapf(err, "listing ServiceInstances in namespace %s", ns)
 		}
-		serviceInstances = append(serviceInstances, svcis.Items...)
+		for _, svci := range svcis.Items {
+			svcClassName := svci.Spec.ServiceClassRef.Name
+			for _, sc := range m.serviceClassByNamespace[ns] {
+				if sc.Name == svcClassName && sc.Spec.ServiceBrokerName == appBrokerServiceClass {
+					serviceInstances = append(serviceInstances, svci)
+				}
+			}
+		}
 	}
 
 	m.serviceInstances = serviceInstances
 
 	return nil
 }
 
+// populateServiceClasses populates the local serviceClassByNamespace map.
+func (m *serviceInstanceManager) populateServiceClasses(namespaces []string) error {
+	serviceClassByNamespace := make(serviceClassByNamespaceMap)
+
+	for _, ns := range namespaces {
+		serviceClass, err := m.cli.ServicecatalogV1beta1().ServiceClasses(ns).List(metav1.ListOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "listing ServiceClasses in namespace %s", ns)
```

Comment:
i really like fmt.Errof("%w", err) as it shows the error message of the wrapped error